examples/data_apis/tensorflow_data_api_tiledb_sparse.ipynb

```diff
@@ -0,0 +1,553 @@
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "We will slice the dataset `user_movie` will be our `x_train` data transformed with one-hot encoding.\n",
+    "So we expect its schema to be the number of ratings as rows and binary columns for users + binary columns\n",
+    "for each item (in our case movies). This will lead to (100000, 2625)\n",
+    "\n",
+    "The target data will be the `ratings`, which will include the ratings and thus will have a shape of (100000,1)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "outputs": [],
+   "source": [
+    "user_movie = data_one_hot[data_one_hot.columns.difference(['rating'])]\n",
+    "ratings = data['rating']"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Data Ingestion\n",
+    "Then we proceed with ingesting `user_movies` into sparse TileDB arrays as our training data and `rating` into dense TileDB\n",
+    "array as our target data. Here, we should point out that besides the\n",
+    "flexibility of TileDB in defining a schema, i.e., multiple dimensions, multiple attributes, compression etc,\n",
+    "we choose to define a simple schema. So, for a numpy array of D number of dimensions we create a dense TileDB array,\n",
+    "with the same number of dimensions, and a single attribute of data type numpy float32."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "outputs": [],
+   "source": [
+    "def get_schema(data: np.array, batch_size: int, sparse: bool) -> tiledb.ArraySchema:\n",
+    "    dims = [\n",
+    "        tiledb.Dim(\n",
+    "            name=\"dim_\" + str(dim),\n",
+    "            domain=(0, data.shape[dim] - 1),\n",
+    "            tile=data.shape[dim] if dim > 0 else batch_size,\n",
+    "            dtype=np.int32,\n",
+    "        )\n",
+    "        for dim in range(data.ndim)\n",
+    "    ]\n",
+    "\n",
+    "    # TileDB schema\n",
+    "    schema = tiledb.ArraySchema(\n",
+    "        domain=tiledb.Domain(*dims),\n",
+    "        sparse=sparse,\n",
+    "        attrs=[tiledb.Attr(name=\"features\", dtype=np.float32)],\n",
+    "    )\n",
+    "\n",
+    "    return schema\n",
+    "\n",
+    "# Let's define an ingestion function\n",
+    "def ingest_in_tiledb(data: np.array, batch_size: int, uri: str, sparse: bool):\n",
+    "    schema = get_schema(data, batch_size, sparse)\n",
+    "\n",
+    "    # Create the (empty) array on disk.\n",
+    "    tiledb.Array.create(uri, schema)\n",
+    "\n",
+    "    # Ingest\n",
+    "    with tiledb.open(uri, \"w\") as tiledb_array:\n",
+    "        idx = np.nonzero(data) if sparse else slice(None)\n",
```

Comment:
This line is failing for me:
```
np.nonzero(user_movie)
...
ValueError: Shape of passed values is (2, 200000), indices imply (100000, 2625)
``` 

Do you have a different pandas version? I'm on 1.2.4

Reply:
Same version `1.2.4`. It is weird still in debug mode I can see the shapes are fine

Reply:
It works if I pass an ndarray instead of Dataframe `np.nonzero(user_movie.values)`

Reply:
There is a cell doing that `execution_count = 8`. Transforms to `numpy_array`. Have you run it?

Reply:
Ah that was it, I missed it. Better pass it to the ingest call (`data=user_movie.to_numpy(), ...`) instead of reassigning the same variables with a new type.